app/controllers/public_controller.rb

```diff
@@ -51,12 +53,20 @@ def commission_histories
     end
   end
 
+  def trent_mode
```

Comment:
This method is related to validators, so it should be placed in Validators Controller, not in Public Controller